src/Application/Commands/ReturnErroneousExcessFees.php

```diff
@@ -82,9 +83,27 @@ protected function doExecute(InputInterface $input, OutputInterface $output): in
                 continue;
             }
 
-            $feeDifferencePence = $fee->amount - $donation->getAmountToDeductFractional();
+            $tipApplicationFeeOffsetPence = 0;
+            // Currently we only support full refunds which update the overall status, and
+            // tip refunds which don't change status but nonetheless set the refund timestamp.
+            // So this `if` clause is a good early indicator that we might have a tip refund
+            // to consider.
+            if ($donation->getDonationStatus() === DonationStatus::Paid && $donation->hasRefund()) {
+                // Stripe PI metadata has the specific original tip as we don't clear that.
+                $paymentIntent = $this->stripeClient->paymentIntents->retrieve($donation->getTransactionId());
+                /** @var numeric-string $tipAmountInMetadata */
+                $tipAmountInMetadata = $paymentIntent->metadata['tipAmount'] ?? '0';
```

Comment:
minor safety improvement:

```suggestion
                $tipAmountInMetadata = $paymentIntent->metadata['tipAmount'] ?? '0';
                \assert(is_string($tipAmountInMetadata) && is_numeric($tipAmountInMetadata));
```